plugin/src/main/java/org/opensearch/ml/processor/MLInferenceIngestProcessor.java

```diff
@@ -0,0 +1,441 @@
+
+        client.execute(MLPredictionTaskAction.INSTANCE, request, new ActionListener<>() {
+
+            @Override
+            public void onResponse(MLTaskResponse mlTaskResponse) {
+                ModelTensorOutput modelTensorOutput = (ModelTensorOutput) mlTaskResponse.getOutput();
+                if (processOutputMap == null || processOutputMap.isEmpty()) {
+                    appendFieldValue(modelTensorOutput, null, DEFAULT_OUTPUT_FIELD_NAME, ingestDocument);
+                } else {
+                    // outMapping serves as a filter to modelTensorOutput, the fields that are not specified
+                    // in the outputMapping will not write to document
+                    Map<String, String> outputMapping = processOutputMap.get(inputMapIndex);
+
+                    for (Map.Entry<String, String> entry : outputMapping.entrySet()) {
+                        // document field as key, model field as value
+                        String newDocumentFieldName = entry.getKey();
+                        String modelOutputFieldName = entry.getValue();
+                        if (ingestDocument.hasField(newDocumentFieldName)) {
+                            throw new IllegalArgumentException(
+                                "document already has field name "
+                                    + newDocumentFieldName
+                                    + ". Not allow to overwrite the same field name, please check output_map."
+                            );
+                        }
+                        appendFieldValue(modelTensorOutput, modelOutputFieldName, newDocumentFieldName, ingestDocument);
+                    }
+                }
+                batchPredictionListener.onResponse(null);
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                batchPredictionListener.onFailure(e);
+            }
+        });
+
+    }
+
+    /**
+     * Retrieves the mapped model input from the IngestDocument and updates the model parameters.
+     *
+     * @param ingestDocument      The IngestDocument object containing the data.
+     * @param modelParameters     The map to store the model parameters.
+     * @param documentFieldName   The name of the field in the IngestDocument.
+     * @param modelInputFieldName The name of the model input field.
+     */
+    private void getMappedModelInputFromDocuments(
+        IngestDocument ingestDocument,
+        Map<String, String> modelParameters,
+        String documentFieldName,
+        String modelInputFieldName
+    ) {
+        // if users used standard dot path, try getFieldPath from document
+        String originalFieldPath = getFieldPath(ingestDocument, documentFieldName);
+        if (originalFieldPath != null) {
+            Object documentFieldValue = ingestDocument.getFieldValue(originalFieldPath, Object.class);
+            String documentFieldValueAsString = toString(documentFieldValue);
```

Comment:
Use `StringUtils.toJson` ?

Reply:
I wrote a toString function in the default method 

https://github.com/opensearch-project/ml-commons/blob/84abae78369b943e634cc22039667ef24c173939/plugin/src/main/java/org/opensearch/ml/processor/ModelExecutor.java#L207

Reply:
will update in the default method